### PR TITLE
T1702 - Donations communication test cases

### PR DESCRIPTION
**Pull request by clementcharmillot**
As the test partner is picked randomly, it might not have any line we can use for displaying. Simply don't restrict the test lines from belonging to the test partner.

### CODE REVIEW

**Comment by ecino** [comment]
Wouldn't it be better to apply a domain to the partner field so that the user can only select partners that have move lines ? Because with your solution, we will select a partner for testing and then the generated communication will be for another one, which will be confusing.

**Comment by clementcharmillot** [comment]
I've implemented your suggestion for the default random selection of a partner. 
But for the partner input, instead of filtering partners, I display an error if the manually picked partner has no invoice of type fund. As a partner having such invoice is not obvious, it might be confusing if it simply doesn't appear in search without explanation when we are searching for a specific one.
Let me know if you'd prefer the other approach.

File: partner_communication_switzerland/models/partner_communication_config.py [hunk 1]
```diff
@@ -141,6 +142,13 @@ def _get_test_objects(self, partner, children=None):
                 )
                 .ids
             )
+            if len(object_ids) == 0:
+                raise UserError(
+                    _(
+                        "The selected partner has no invoice line but this test case "
+                        "requires one. Please, pick a partner with an invoice line."
```
Comment (ecino) [inline partner_communication_switzerland/models/partner_communication_config.py]:
```suggestion
                        "The selected partner is not a donor but this test case "
                        "requires one. Please, pick a donor."
```

**Review by ecino — CHANGES_REQUESTED** [review]
OK, but the error message will be more explicit to our staff like that.

**Pull request merged**